topi/tests/python/test_topi_depthwise_conv2d_back_input.py

```diff
@@ -45,6 +41,57 @@ def check_device(device):
         # build the kernels
         f = tvm.build(schedule, [Filter, Out_grad, In_grad], device)
 
+        # Use memoize, pickle the test data for next time use.
+        @memoize("topi.tests.test_topi_depthwise_conv2d_backward_input")
```

Comment:
This part is still wrong. The error is the contrib/pickle_memoize.py, line 80. allow_types error?

Reply:
The restriction is that we can only capture int, str, float or tuple of them, to be able to hash the input correctly. Please refer to how we do it in the depthwise conv testcase. Specifically, we will need to get the shape outside this closure first instead of call get_const_tuple inside the function.

Reply:
Thanks. I got it working. The shape and type has to be defined outside the get_ref_data (just a reference for future usage).